plugins/techdocs-backend/src/service/router.test.ts

```diff
@@ -94,34 +146,19 @@ describe('createRouter', () => {
     discovery.getBaseUrl.mockImplementation(async type => {
       return `http://backstage.local/api/${type}`;
     });
-
-    const outOfTheBoxRouter = await createRouter({
-      preparers,
-      generators,
-      publisher,
-      config: new ConfigReader({}),
-      logger: getVoidLogger(),
-      discovery,
-    });
-    const recommendedRouter = await createRouter({
-      publisher,
-      config: new ConfigReader({}),
-      logger: getVoidLogger(),
-      discovery,
-    });
-
-    app = express();
-    app.use(outOfTheBoxRouter);
-    app.use('/recommended', recommendedRouter);
-    app.use(errorHandler());
+    MockedConfigReader.prototype.getOptionalNumber.mockImplementation(key =>
```

Comment:
This prototype usage confounds me 😄 Never seen that before.

Anyway, mocking the config reader sounds inadvisable. Just giving in actual config instances with `new MockConfigApi({ techdocs: { cache: { ttl: 1 }}})` etc feels more appropriate.

Reply:
Agreed! This test suite was already using this prototype pattern so I just stuck with that. Happy to do this refactor with the follow up work we discussed in https://github.com/backstage/backstage/pull/8791#discussion_r780117326